src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rest/QueryMetricsRequestHandler.java

```diff
@@ -68,7 +68,17 @@ public QueryMetricsRequestHandler() {
     @Override
     public void handle(HttpExchange exchange) throws IOException {
         String requestMethod = exchange.getRequestMethod();
-        Map.Entry<Long, MetricsDB> dbEntry = ReaderMetricsProcessor.current.getMetricsDB();
+        ReaderMetricsProcessor mp = ReaderMetricsProcessor.getInstance();
+        if (mp == null) {
```

Comment:
nit: will there be ever a case, where this is null?

Reply:
It can be null if the metricsProcessor thread was very slow to initialize, but the httpserver was initialized and got a request from a client.